api/pkg/s3/bucketacl.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"encoding/xml"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/emicklei/go-restful"
+	"github.com/journeymidnight/yig/helper"
+	"github.com/opensds/multi-cloud/api/pkg/common"
+	"github.com/opensds/multi-cloud/api/pkg/s3/datatype"
+	s3error "github.com/opensds/multi-cloud/s3/error"
+	pb "github.com/opensds/multi-cloud/s3/proto"
+	log "github.com/sirupsen/logrus"
+)
+
+func (s *APIService) BucketAclPut(request *restful.Request, response *restful.Response) {
+	bucketName := strings.ToLower(request.PathParameter(common.REQUEST_PATH_BUCKET_NAME))
+	log.Infof("received request: PUT bucket[name=%s] acl\n", bucketName)
+
+	var err error
+	var acl datatype.Acl
+	var policy datatype.AccessControlPolicy
+	if _, ok := request.Request.Header[common.REQUEST_HEADER_ACL]; ok {
+		acl, err = getAclFromHeader(request)
+		if err != nil {
+			WriteErrorResponse(response, request, err)
+			return
+		}
+	} else {
+		aclBuffer, err := ioutil.ReadAll(io.LimitReader(request.Request.Body, 1024))
```

Comment:
Hi @sunfch why it's limited to 1024, could you please add some comment to explain here?

Reply:
@sfzeng thanks for your reviewing,  because we can only support canned acl, the body of acl request must are less 1024.  The 1024 is my guess, I will use const declare replace it and comment it?

Reply:
@kumarashit ,  you are right, the policy in http body is parsed in else branch.  because ACL is no longer recommended to use,  and Cannel ACL is most used usually,   we only want to support  the policy that is able to be parsed to Canned ACL.

Reply:
Ok